drake/systems/controllers/LQR.h

```diff
@@ -17,15 +17,11 @@ namespace Drake {
     using namespace std;
     using namespace Eigen;
 
-    // todo: clean this up
-    typedef TaylorVard<-1> AutoDiffType;
-    VectorXd xu(num_states+num_inputs);
-    xu << toEigen(x0), toEigen(u0);
-    TaylorVecXd xu_taylor = initTaylorVecXd(xu);
-    typename System::template StateVector<AutoDiffType> x_taylor(xu_taylor.head(num_states));
-    typename System::template InputVector<AutoDiffType> u_taylor(xu_taylor.tail(num_inputs));
-
-    auto xdot = autoDiffToGradientMatrix(toEigen(sys.dynamics(AutoDiffType(0),x_taylor,u_taylor)));
+    auto autodiff_args = initializeAutoDiffArgs(toEigen(x0), toEigen(u0));
```

Comment:
So this is what the new stuff in Gradient.h allows you to do. Given a bunch of Eigen matrix arguments, construct a tuple of corresponding AutoDiff matrices, with values and derivatives properly initialized.